concordia/language_model/local_models.py

```diff
@@ -0,0 +1,307 @@
+from concordia.language_model import language_model
+from concordia.utils import measurements as measurements_lib
+
+from langchain.llms import Ollama
+from langchain.callbacks.manager import CallbackManager
+from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler
+
+from collections.abc import Collection, Sequence
+from string import Template
+from typing_extensions import override
+
+import logging
+import re
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+_MAX_MULTIPLE_CHOICE_ATTEMPTS = 5
+_MAX_SAMPLE_TEXT_ATTEMPTS = 5
+
+# PROMPT_TEMPLATE = Template("""<s>[INST] <<SYS>>\n$system_message\n<</SYS>>\n$message[/INST]""")
+PROMPT_TEMPLATE = Template("$system_message\n\n$message")
+
+# TEXT_SYSTEM_MESSAGE = """Your task is to follow instructions and answer questions correctly and concisely."""
+TEXT_SYSTEM_MESSAGE = ""
+
+MULTIPLE_CHOICE_SYSTEM_MESSAGE = """The following is a multiple choice question.
+You must always respond with one of the possible choices.
+You may not respond with anything else.
+"""
+# MULTIPLE_CHOICE_SYSTEM_MESSAGE = ""
+
+class OllamaLanguageModel(language_model.LanguageModel):
```

Comment:
How is this different from the existing ollama model at HEAD? Wouldn't it be confusing to have two of them? Wouldn't it be better to add whatever enhancements are here into the existing model?

Reply:
Thanks. This is strange, there should have only been the random premise_generator since I did not touch local_models here.